README.md

```diff
@@ -1,44 +1,93 @@
-# smt
+# Sparse Merkle Tree (smt)
 
-A Go library that implements a Sparse Merkle tree for a key-value map. The tree implements the same optimisations specified in the [Libra whitepaper][libra whitepaper], to reduce the number of hash operations required per tree operation to O(k) where k is the number of non-empty elements in the tree.
+A Go library that implements a Sparse Merkle Tree for a key-value map. The tree implements the same optimisations specified in the [Jellyfish Merkle Tree whitepaper][jmt whitepaper] originally designed for the [Libra blockchain][libra whitepaper]. It reduces the number of hash operations required per tree operation to `O(k)` where `k` is the number of non-empty elements in the tree.
```

Comment:
This library is not based on JMT, just an optimized SMT. The JMT was introduced after the original Libra whitepaper, which is basically an SMT but with improvements on the storage layer (storing it as a 16-ary tree, etc).

Reply:
Seems like a JMT is just an SMT with r=16 + RocksDB + versioned keys.

<img width="988" alt="Screen Shot 2022-10-31 at 6 39 04 PM" src="https://user-images.githubusercontent.com/1892194/199140010-58703590-e5b9-48db-aab0-7e3f34be1edd.png">